src/views/MainView.js

```diff
@@ -55,10 +68,155 @@
       event.stopPropagation();
       event.stopImmediatePropagation();
       $('.nav .menu-btn').addClass('hidden');
+      if ($.os.ios || $.os.android || $.os.bb10) {
+        $('.nav .share-btn').addClass('hidden');
+        $('.nav .copy-btn').addClass('hidden');
+      } else {
+        $('.nav .export-btn').addClass('hidden');
+      }
       $(".fab").addClass("shrunken");
       window.app.navigator.pushView(window.app.SettingsView, {},
         window.app.defaultEffect);
     },
+    addFieldFromEntry: function(entry, fields){
+      entry.items.forEach(function(item) {
+        var field = item.key;
+        if (fields.indexOf(field) === -1) {
+          fields.push(field);
+        }
+      });
+    },
+    getCsvFields: function(entries) {
+      var fields = ['Entry Type', 'Label'];
+      for (var index in entries){
+        var entry = entries[index];
+        this.addFieldFromEntry(entry, fields);
+      }
+      return fields;
+    },
+    addDataFromEntry: function(entry, fields){
+      var entry_data = {};
+      entry.items.forEach(function(item) {
+        var field = item.key;
+        entry_data[field] = item.value;
+      });
+      entry_data['Entry Type'] = entry.type;
+      entry_data['Label'] = entry.label;
+      return entry_data;
+    },
+    getCsvData: function(entries, fields) {
+      var data = [];
+      for (var index in entries){
+        var entry = entries[index];
+        data.push(this.addDataFromEntry(entry, fields));
+      }
+      return data;
+    },
+    writeCordovaFile: function(fileName, data) {
+      data = JSON.stringify(data, null, '\t');
+      var promise = $.Deferred();
+      window.resolveLocalFileSystemURL(cordova.file.cacheDirectory, function (directoryEntry) {
+          directoryEntry.getFile(fileName, { create: true }, function (fileEntry) {
+              fileEntry.createWriter(function (fileWriter) {
+                  var blob = new Blob([data], { type: 'text/csv' });
+                  fileWriter.write(blob);
+                  promise.resolve(fileEntry.fullPath);
+              }, console.log);
+          }, console.log);
+      }, console.log);
+      return promise;
+    },
+    saveCsv: function(csv){
+      var type = 'text/csv';
+      var file = new Blob([csv], {type: type});
+      var a = document.createElement("a");
+      var url = window.URL.createObjectURL(file);
+      a.href = url;
+      a.target="_blank";
+      a.download = 'export.csv';
+      a.hidden = true;
+      document.body.appendChild(a);
+      a.click();
+      document.body.removeChild(a);
+      window.URL.revokeObjectURL(url);
```

Comment:
Right, ok, nono, sorry, that's fine. I just kind of hated that it needed to be so hacky. Please do add a comment explaining that we need to support 0.11.5 for now and that's why it's the way it is?